content/spin/variables.md

```diff
@@ -0,0 +1,205 @@
+title = "Configuration Variables"
+template = "spin_main"
+date = "2022-06-14T00:22:56Z"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main//content/spin/variables.md"
+
+---
+- [Using the Config SDK to Get Variables](#using-the-config-sdk-to-get-variables)
+
+Spin supports dynamic application variables. Instead of being static, their values can be updated without modifying the application, creating a simpler experience for rotating secrets, updating API endpoints, and more. 
+
+These variables are defined in a Spin application manifest (in the `[variables]` section) and are provided by a [configuration provider](/spin/dynamic-configuration.md#custom-config-providers). When running Spin locally, the configuration provider can be Vault for secrets or host environment variables. Refer to the [dynamic configuration documentation](/spin/dynamic-configuration.md) to learn how to configure variables locally.
+
+## Using the Config SDK to Get Variables
+
+The Spin SDK surfaces the Spin Config interface to your language. The [interface](https://github.com/fermyon/spin/blob/main/wit/ephemeral/spin-config.wit) consists of one operation:
+
+| Operation  | Parameters                          | Returns             | Behavior |
+|------------|-------------------------------------|---------------------|----------|
+| `get-config`    | Variable name  | Variable value    | Gets the value of the variable from the configured provider |
+
+In order to get a variable, it must be declared in an application's manifest (`spin.toml`). For example, say an application needs to access a secret. A variable named `password` could be added to the manifest and set as required with `required = true` since there is no reasonable default value for a secret. To do this, add a top-level `[variables]` section to the application manifest (`spin.toml`) and declare the variable within it.
+
+<!-- @selectiveCpy -->
+```toml
+# Add this above the [component] section
+[variables]
+password = { required = true }
+```
+
+To surface the variable to a component, add a `[component.config]` section in the component and specify the variable within it. Instead of statically assigning the value of the config variable, dynamically reference the variable with [mustache](https://mustache.github.io/)-inspired string templates. Only components that explicitly use the variables in their configuration section will get access to them. This enables only exposing variables and secrets to the desired components of an application.
+
+```toml
+# Add this below the [component.build] section
+[component.config]
+password = "\{{ password }}"
```

Comment:
It would be really good to use an example where application variables and component config values have different names and values - I think this would help illustrate the distinction and the flow.  I am not sure of a good example though - maybe database host and access token as variables, and connection string as the config value?  (Unfortunately I suspect our real connection strings need more inputs than that...)

Reply:
thats a good example but i think i am already getting a little verbose with the examples here. I could change it to something mocked with a fake db host and password but i am not sure i could add clean working examples

Reply:
No worries then, ship it as is.  We can always iterate if and when we feel motivated.

Reply:
I added a static mapping to the "Adding Variables to an Application" section but am not using that config variable in the example implementations